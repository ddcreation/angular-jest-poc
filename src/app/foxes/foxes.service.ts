import { Injectable } from '@angular/core';

export interface Fox {
  id: number;
  name: string;
}

@Injectable({
  providedIn: 'root',
})
export class FoxesService {
  public foxes: Fox[] = [];

  public add(foxname: string): Fox {
    const newFox = {
      id: this.foxes.length + 1,
      name: foxname,
    };

    this.foxes.push(newFox);
    return newFox;
  }
}
