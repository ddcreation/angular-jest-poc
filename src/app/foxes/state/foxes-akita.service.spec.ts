import { TestBed } from '@angular/core/testing';

import { FoxesAkitaService } from './foxes-akita.service';
import { FoxesStore } from './foxes.store';
import { FoxesQuery } from './foxes.query';

describe('FoxesAkitaService', () => {
  let service: FoxesAkitaService;
  let query: FoxesQuery;
  let store: FoxesStore;

  beforeEach(() => {
    TestBed.configureTestingModule({
      providers: [FoxesStore],
    });
    service = TestBed.inject(FoxesAkitaService);
    store = TestBed.inject(FoxesStore);
    query = new FoxesQuery(store);
  });

  it('should be created', () => {
    expect(service).toBeTruthy();
  });

  it('should have empty initial state', () => {
    expect(query.getValue()).toMatchSnapshot('initialState');
  });

  it('should add a fox', () => {
    service.add('New fox');

    const entities = query.getValue().entities;
    expect(entities[Object.keys(entities)[0]]).toMatchSnapshot({
      id: expect.any(Number),
    });
  });

  it('should reset', () => {
    service.reset();

    expect(query.getValue()).toMatchSnapshot('initialState');
  });
});
