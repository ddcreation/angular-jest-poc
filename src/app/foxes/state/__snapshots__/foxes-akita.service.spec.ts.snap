// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoxesAkitaService should add a fox 1`] = `
Object {
  "id": Any<Number>,
  "name": "New fox",
}
`;

exports[`FoxesAkitaService should have empty initial state: initialState 1`] = `
Object {
  "entities": Object {},
  "error": null,
  "ids": Array [],
  "loading": true,
}
`;

exports[`FoxesAkitaService should reset: initialState 1`] = `
Object {
  "entities": Object {},
  "error": null,
  "ids": Array [],
  "loading": true,
}
`;
